File: template/template/src/api/ReactNativeTHEOplayerConnector.ts
import type { THEOplayer } from 'react-native-theoplayer';
import { ReactNativeTHEOplayerConnectorAdapter } from '../internal/ReactNativeTHEOplayerConnectorAdapter';
import type { ReactNativeTHEOplayerConnectorConfiguration } from "./ReactNativeTHEOplayerConnectorConfiguration";

export class ReactNativeTHEOplayerConnector {

  private connectorAdapter: ReactNativeTHEOplayerConnectorAdapter

  /**
   * Create a new connector instance.
   *
   * @param player THEOplayer instance.
   * @param config configuration object.
   */
  constructor(player: THEOplayer, config: ReactNativeTHEOplayerConnectorConfiguration) {
    this.connectorAdapter = new ReactNativeTHEOplayerConnectorAdapter(player, config);
  }

  /**
   * Destroy connector.
   */
  destroy(): void {
    this.connectorAdapter.destroy();
  }
}

File: template/template/android/src/main/java/com/ReactNativeTHEOplayerConnector/ReactNativeTHEOplayerConnector.kt
package com.ReactNativeTHEOplayerConnector

import android.content.Context
import com.theoplayer.android.api.player.Player

class ReactNativeTHEOplayerConnector(
    private val context: Context,
    private val player: Player,
    private val config: ReactNativeTHEOplayerConnectorConfiguration
) {

    init {
        // Initialize connector
    }

    fun destroy() {
        // Destroy connector
    }
}